K2Field.K2NE.ServiceBroker/ServiceObjects/EmailTemplate/EmailTemplateSO.cs

```diff
@@ -0,0 +1,256 @@
+using System.Data;
+using System.Linq;
+using System.Text;
+using K2Field.K2NE.ServiceBroker.Helpers;
+using SourceCode.SmartObjects.Client;
+using SourceCode.SmartObjects.Services.ServiceSDK.Objects;
+using SourceCode.Workflow.Client;
+using SourceCode.SmartObjects.Services.ServiceSDK.Types;
+using MethodType = SourceCode.SmartObjects.Services.ServiceSDK.Types.MethodType;
+
+namespace K2Field.K2NE.ServiceBroker.ServiceObjects.EmailTemplate
+{
+    public class EmailTemplateSO : ServiceObjectBase
+    {
+        public EmailTemplateSO(K2NEServiceBroker worklistAPI) : base(worklistAPI) { }
+
+        public override string ServiceFolder
+        {
+            get
+            {
+                return "Email Template";
+            }
+        }
+
+        public override List<ServiceObject> DescribeServiceObjects()
+        {
+            ServiceObject so = Helper.CreateServiceObject("EmailTemplate", "ServiceObject that provides email template functionality.");
+
+            so.Properties.Add(Helper.CreateProperty(Constants.SOProperties.EmailTemplate.EmailBody, SoType.Memo, "EmailBody"));
+            so.Properties.Add(Helper.CreateProperty(Constants.SOProperties.EmailTemplate.EmailSubject, SoType.Text, "EmailBody"));
+            so.Properties.Add(Helper.CreateProperty(Constants.SOProperties.EmailTemplate.TemplateLanguage, SoType.Text, "TemplateLanguage"));
+            so.Properties.Add(Helper.CreateProperty(Constants.SOProperties.EmailTemplate.TemplateName, SoType.Text, "TemplateName"));
+            so.Properties.Add(Helper.CreateProperty(Constants.SOProperties.EmailTemplate.EmailTemplateId, SoType.Number, "Email Template Id"));
+            
+
+            Method mGetEmailTemplate = Helper.CreateMethod(Constants.Methods.EmailTemplate.GetEmailTemplate, "Get Email Template by name.", MethodType.Read);
+            mGetEmailTemplate.InputProperties.Add(Constants.SOProperties.EmailTemplate.TemplateName);
+            mGetEmailTemplate.InputProperties.Add(Constants.SOProperties.EmailTemplate.TemplateLanguage);
+            mGetEmailTemplate.Validation.RequiredProperties.Add(Constants.SOProperties.EmailTemplate.TemplateName);
+            mGetEmailTemplate.Validation.RequiredProperties.Add(Constants.SOProperties.EmailTemplate.TemplateLanguage);
+            mGetEmailTemplate.ReturnProperties.Add(Constants.SOProperties.EmailTemplate.TemplateName);
+            mGetEmailTemplate.ReturnProperties.Add(Constants.SOProperties.EmailTemplate.TemplateLanguage);
+            mGetEmailTemplate.ReturnProperties.Add(Constants.SOProperties.EmailTemplate.EmailBody);
+            mGetEmailTemplate.ReturnProperties.Add(Constants.SOProperties.EmailTemplate.EmailSubject);
+            mGetEmailTemplate.MethodParameters.Add(Helper.CreateParameter(Constants.SOProperties.EmailTemplate.ProcessInstanceId, SoType.Number, false, "Process Instance Id"));
+            foreach (var param in GetCustomParamaters(EmailTemplateCustomParameters))
```

Comment:
Yes, there is. I am not returning them. Why do I need the properties if I am not returning them? I only use them to change the placeholders. Also I would like to avoid possible K2 internal filtering, cause it never happens with parameters.